imap_processing/lo/l0/science_direct_events.py

```diff
@@ -0,0 +1,202 @@
+
+
+@dataclass
+class ScienceDirectEvents(LoL0):
+    """L1A Science Drect Events data.
+
+    The Science Direct Events class handles the parsing and
+    decompression of L0 to L1A data.
+
+    Attributes
+    ----------
+    SHCOARSE : int
+        Spacecraft time.
+    COUNT: int
+        Number of direct events.
+    CHKSUM: int
+        Checksum for the packet.
+    DATA: bitstring.Bits
+        Compressed TOF Direct Event time tagged data.
+    TOF0: int
+        Time of Flight 0 value for direct event.
+    TOF1: int
+        Time of Flight 1 value for direct event.
+    TOF2: int
+        Time of Flight 2 value for direct event.
+    TOF3: int
+        Time of Flight 3 value for direct event.
+    TIME: int
+        time tag for the direct event
+    ENERGY: int
+        energy of the direct event ENA.
+    POS: int
+        Stop position for the direct event. There are 4 quadrants
+        on the at the stop position.
+    CKSM: int
+        This is checksum defined relative to the TOFs
+        condition for golden triples. If golden triples are below
+        a certain threshold in checksum it's considered golden, otherwise,
+        it's considered a silver triple. This is important for the compression
+        for golden triples because it's used to recover TOF1 because
+        compression scheme to save space on golden triples doesn't send
+        down TOF1 so it's recovered on the ground using the checksum
+    case_number: int
+        The compression case number for the direct event. The case number
+        determines how the bits are arranged in the compressed data.
+    tof_calculation_binary: dict
+        Binary used to determine which TOF coefficients should be used
+        for decompressing the binary.
+    tof_decoder: list
+        Shows how the fields in the binary are split up by bit length and order.
+    remaining_bits: dict
+        The TOF coefficients that should be used for decompression
+    parsed_bits: dict
+        The binary bits split up by TOF field.
+
+    Methods
+    -------
+    __init__(packet, software_vesion, packet_file_name):
+        Constructor, uses the CCSDS packet, version of the software, and
+        the name of the packet file to parse and store information about
+        the Direct Event packet data.
+    decompress_data():
+        Decompresses the Science Direct Event TOF data.
+
+    """
+
+    SHCOARSE: int
+    COUNT: int
+    DATA: bitstring.Bits
+    CHKSUM: int
+    TOF0: float
+    TOF1: float
+    TOF2: float
+    TOF3: float
+    TIME: float
+    ENERGY: float
+    POS: float
+    case_number: int
+    tof_calculation_binary: dict
+    tof_decoder: list
+    remaining_bits: dict
+    parsed_bits: list
+
+    def __init__(self, packet, software_version: str, packet_file_name: str):
+        super().__init__(software_version, packet_file_name, CcsdsData(packet.header))
+        self.parse_data(packet)
+
+    def decompress_data(self):
```

Comment:
Where is this method used? It's a little confusing to me that it's not called as part of `__init__`. Is there a case where you would use the data without calling `decompress_data`?

Reply:
I was thinking it would be used in the decom code (that I haven't put out for PR yet). as a separate method call. So it'd be like:
```python
sci_de = ScienceDirectEvent(packet, software_version, packet_name)
sci_de.decompress_data()
```

But I guess there would be no scenario where I'd create a sci de object and not decompress, so I'll put it in the __init__ or __post_init__, whatever one I end up using.

Reply:
Yeah if you're going to always apply it, I definitely vote for putting it in `__init__` or `__post_init__`